### DISCUSSION TITLE
[Projects Beta]: Date filters

### DISCUSSION THREAD

**Discussion by noamross** [community/community#5953] [Projects and Issues]
It would be great to be able to filter by date, including by relative dates.  I'd like to be able to filter a view by a date field like `due date < (today + 2 weeks)`.

**Comment by bewuethr** [comment]
Or, similar to PR and issue filters in search, with a range like `Created at:>=2021-09-01`

**Reply by djparkopedia** [reply]
def this - same syntax as we have in the standard issues view would be great!

**Reply by polevaultweb** [reply]
Also the ability to filter by issue closed date too

**Comment by mdtusz** [comment]
Any movement on this feature? This seems like a fairly critical feature to omit.

**Reply by bewuethr** [reply]
The kind of filter asked for in the question are [now available](https://github.blog/changelog/2022-08-18-github-issues-august-18th-update/#date-date-filter-improvements), it seems.

Edit: or maybe not exactly... but close.

**Reply by mdtusz** [reply]
Iterations seems to work, but dates don't - that documentation seems to be either incorrect or just was published before the feature has rolled out.

`created` seems to be a reserved name and you can't add it as a column, so I'm assuming the data exists and is there but the filtering mechanism just isn't equipped to filter by created dates yet.

Hopefully it gets resolved soon!

**Reply by golgor** [reply]
> The kind of filter asked for in the question are [now available](https://github.blog/changelog/2022-08-18-github-issues-august-18th-update/#date-date-filter-improvements), it seems.
> 
> Edit: or maybe not exactly... but close.

This seems to still be the case... it is not possible to use the date-field. :(

**Comment by vhartikainen** [comment]
Is this really missing from Projects still? For instance:
1) Create issue
2) In Github issues: created:>2023-11-10 

But in projects:
1) Create issue
2) Create a custom date type field: created_date
3) Copy the created date -> created_date field
4) Filter projects board view with: created_date:>2023-11-10

This seems wrong in so many ways :)

**Comment by johanvaneck** [comment]
Any updates on this?

**Comment by michaelklopf** [comment]
Hey, I wonder if this will come at anytime, because it's weird that I can't find all the tasks added last year or this year.

**Comment by gauthierpetetin** [comment]
Hello! Is it on Github Project's roadmap to add the ability to filter by date?
I'd like to be able to display a list of issues with this kind of filter:
`createdAt:>2024-04-01`

**Comment by major0** [comment]
Really, it seems really odd that basically all date handling is missing:
- Filter based on time range, such as created this week.
- Sort by age, such as displaying the oldest tasks, or finding tasks that have been in the backlog for a year.

I imagine I could add some automation to do this, but it is such a basic concept it is hard for me to comprehend why it isn't already part of the system.

**Comment by HughParry** [comment]
Any movement on this? Pretty critical feature

**Comment by marksverdhei** [comment]
It is 2025, How do we still not have this?

**Comment by per1234** [comment]
The capability requested by OP is now available:

https://docs.github.com/issues/planning-and-tracking-with-projects/customizing-views-in-your-project/filtering-projects#filtering-iteration-and-date-fields-using-keywords

So if you create a new field with name "due date" with "Date" type, you can then use the following query to see the items which have a value in that field less than two weeks in the future:

```text
due-date:<@today+2w
```

---

The capability for filtering by a built-in creation date field requested by the other commenters is still not available for projects. You could achieve this by adding a custom field for that purpose, but you would need to manually set the value of the field for each item, which is inconvenient.

Not what the commenters requested, but in case it might be of interest to others who find this discussion while searching for the information, I'll mention that there is a capability for filtering by the date the item was updated via the built-in `updated` field, for example to see the items that have been updated within the last week:

```text
updated:>@today-1w
```

The [creation of an item is considered an update](https://docs.github.com/issues/planning-and-tracking-with-projects/customizing-views-in-your-project/filtering-projects#filtering-for-when-an-item-was-last-updated), so for some specific use cases this might serve as an alternative to the missing built-in `creation` field.

**Reply by soura-b** [reply]
The range of use cases where `updated` is a meaningful substitute for `created` is quite narrow. A recent comment on an issue thread, or a label change, is all it takes for the `updated` to, well, update. Finding items created before a particular date is still onerous.
